src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -3215,7 +3215,7 @@ BinaryOpExpression(Expr *clause, Node **leftOperand, Node **rightOperand)
 		}
 		if (rightOperand != NULL)
 		{
-			*leftOperand = NULL;
+			*rightOperand = NULL;
```

Comment:
Any way to get this covered by tests? Because this seems like it fixes an actual bug.

Reply:
Unfortunately, no, this is a generic API, in our case, both the callers pass addresses and check them if and only if the routine returns TRUE. But in this scenario, we return FALSE, so we never hit this bug as operand(s) are not used.

```
Node *leftOperand;
Node *rightOperand;
        if (!BinaryOpExpression(clause, &leftOperand, &rightOperand))
               return false;
```
```
Node *leftOperand;
Node *rightOperand;
        if (!BinaryOpExpression((Expr *) opClause, &leftOperand, &rightOperand))
                return false;
```

Reply:
Wait, I think your question is different, it's about code coverage and not why a unit test didn't hit this bug? I will be surprised we don't have a unit test where dist key is not a binary operator expression, let me check.

Reply:
To exercise this code, either one of the following should be true

`        if (!is_opclause(clause) || list_length(((OpExpr *) clause)->args) != 2)
`
All the callers are calling with OpExpr, so the first operand is always false. We need an operator with args>2 to use on dist-key. Tried some obvious operators including `like`, but no luck.

Reply:
`select id from source where id = @  '-4.5' ;`
`select id from source where id = 2^3;
`

In a WHERE clause we need to have a binary operator, though operand(s) may not simple. I will continue to explore.

Reply:
Argument of WHERE must be type boolean, tested manually various ways, either we are ending with non OpExpr, such as T_BooleanTest or OpExpr with two arguments. We have to find a novel way of testing an OpExpr with arguments !=2